src/autogenerated/java/net/minestom/server/particle/ParticleType.java

```diff
@@ -0,0 +1,292 @@
+        Registry.PARTICLE_REGISTRY.register(FISHING);
+        Registry.PARTICLE_REGISTRY.register(FLAME);
+        Registry.PARTICLE_REGISTRY.register(SOUL_FIRE_FLAME);
+        Registry.PARTICLE_REGISTRY.register(SOUL);
+        Registry.PARTICLE_REGISTRY.register(FLASH);
+        Registry.PARTICLE_REGISTRY.register(HAPPY_VILLAGER);
+        Registry.PARTICLE_REGISTRY.register(COMPOSTER);
+        Registry.PARTICLE_REGISTRY.register(HEART);
+        Registry.PARTICLE_REGISTRY.register(INSTANT_EFFECT);
+        Registry.PARTICLE_REGISTRY.register(ITEM);
+        Registry.PARTICLE_REGISTRY.register(ITEM_SLIME);
+        Registry.PARTICLE_REGISTRY.register(ITEM_SNOWBALL);
+        Registry.PARTICLE_REGISTRY.register(LARGE_SMOKE);
+        Registry.PARTICLE_REGISTRY.register(LAVA);
+        Registry.PARTICLE_REGISTRY.register(MYCELIUM);
+        Registry.PARTICLE_REGISTRY.register(NOTE);
+        Registry.PARTICLE_REGISTRY.register(POOF);
+        Registry.PARTICLE_REGISTRY.register(PORTAL);
+        Registry.PARTICLE_REGISTRY.register(RAIN);
+        Registry.PARTICLE_REGISTRY.register(SMOKE);
+        Registry.PARTICLE_REGISTRY.register(SNEEZE);
+        Registry.PARTICLE_REGISTRY.register(SPIT);
+        Registry.PARTICLE_REGISTRY.register(SQUID_INK);
+        Registry.PARTICLE_REGISTRY.register(SWEEP_ATTACK);
+        Registry.PARTICLE_REGISTRY.register(TOTEM_OF_UNDYING);
+        Registry.PARTICLE_REGISTRY.register(UNDERWATER);
+        Registry.PARTICLE_REGISTRY.register(SPLASH);
+        Registry.PARTICLE_REGISTRY.register(WITCH);
+        Registry.PARTICLE_REGISTRY.register(BUBBLE_POP);
+        Registry.PARTICLE_REGISTRY.register(CURRENT_DOWN);
+        Registry.PARTICLE_REGISTRY.register(BUBBLE_COLUMN_UP);
+        Registry.PARTICLE_REGISTRY.register(NAUTILUS);
+        Registry.PARTICLE_REGISTRY.register(DOLPHIN);
+        Registry.PARTICLE_REGISTRY.register(CAMPFIRE_COSY_SMOKE);
+        Registry.PARTICLE_REGISTRY.register(CAMPFIRE_SIGNAL_SMOKE);
+        Registry.PARTICLE_REGISTRY.register(DRIPPING_HONEY);
+        Registry.PARTICLE_REGISTRY.register(FALLING_HONEY);
+        Registry.PARTICLE_REGISTRY.register(LANDING_HONEY);
+        Registry.PARTICLE_REGISTRY.register(FALLING_NECTAR);
+        Registry.PARTICLE_REGISTRY.register(ASH);
+        Registry.PARTICLE_REGISTRY.register(CRIMSON_SPORE);
+        Registry.PARTICLE_REGISTRY.register(WARPED_SPORE);
+        Registry.PARTICLE_REGISTRY.register(DRIPPING_OBSIDIAN_TEAR);
+        Registry.PARTICLE_REGISTRY.register(FALLING_OBSIDIAN_TEAR);
+        Registry.PARTICLE_REGISTRY.register(LANDING_OBSIDIAN_TEAR);
+        Registry.PARTICLE_REGISTRY.register(REVERSE_PORTAL);
+        Registry.PARTICLE_REGISTRY.register(WHITE_ASH);
+    }
+
+    @NotNull
+    private final NamespaceID id;
+
+    @NotNull
+    private final BiFunction<ParticleType<T>, @Nullable String, T> dataReader;
+
+    protected ParticleType(@NotNull NamespaceID id,
```

Comment:
Since `ParticleType` contains both the type and the data, wouldnt it make more sense to keep the `Particle` name?

Reply:
The ParticleType doesn't contain the data but it does contain the reader to get the data. I decided to do that because you can't get the generic class (like particleType.getDataType().getReader()). The reader is essentially just a "pointer" to the data class.